src/Microsoft.OData.Client/ALinq/DataServiceQueryProvider.cs

```diff
@@ -322,7 +322,16 @@ private TElement ParseAggregateSingletonResult<TElement>(QueryResult queryResult
                             if (entry != null && properties?.Any() == true)
                             {
                                 ODataProperty aggregationProperty = properties.First();
-                                ODataUntypedValue untypedValue = aggregationProperty.Value as ODataUntypedValue;
+
+                                if (aggregationProperty.Value == null)

```

Comment:
Is there a possibility that aggregationProperty can be null
```suggestion
                                if (aggregationProperty?.Value == null)
```

Reply:
Given the use of `properties?.Any() == true` and `properties.First()`, I don't think there's room for `aggregationProperty` to be null

Reply:
+1